pyne/mesh.py

```diff
@@ -1008,9 +1046,19 @@ def tag(self, name, value=None, tagtype=None, doc=None, size=None,
                                  'or dtype'.format(name))
             else:
                 tagtype = MetadataTag
+
         if tagtype is NativeMeshTag or tagtype.lower() == 'nat_mesh':
+            # storage_type works for only NativeMeshTag
+            if storage_type is None or storage_type.lower() == 'dense':
+                storage_type = types.MB_TAG_DENSE
+            elif storage_type == types.MB_TAG_SPARSE or \
```

Comment:
I think we only want to check if the `storage_type` is equal to `'sparse'` here.